### PR TITLE
zh-CN: fix fuzzy sentences

### PR DESCRIPTION
**Pull request by SketchK**
Hi, @henrif75 
The diff looks normal now!
Thank for your work!

### CODE REVIEW

**Comment by SketchK** [comment]
Based on what happened before, the other reviewers were so slow to respond that my PR never got merged in.
Do you have any suggestions for this situation?
Why don't I finish the translation myself and you help me approve it?  @henrif75

**Comment by SketchK** [comment]
> Based on what happened before, the other reviewers were so slow to respond that my PR never got merged in. Do you have any suggestions for this situation? Why don't I finish the translation myself and you help me approve it? @henrif75 I'll make sure of the quality of the translation myself.

If this proposal is acceptable, I think I'll be able to translate the entire Chinese version in about a week's time.
@henrif75 @mgeisler

**Comment by mgeisler** [comment]
Hey @SketchK, as you can imagine, neither @henrif75 nor myself speak Chinese... so we've been reliant on the help of others like yourself who do. We have found those volunteers both internally at Google and externally.

I'm sorry that your PRs have been waiting — I guess the people who helped out so far have become busy with other things. I'll try writing them and see if we can revive this.

I'm against blindly accepting the translation since I've seen myself how helpful it can be to have another pair of eyes look things over. I make a ton of typos when I write anything, so I've benefited from the reviews of the Danish translation :smile: 

However, if we cannot find anyone to help here, then merging your PRs is definitely better than not. Let us see if we can reach the old translators in the next few days?

@henrif75, related to this, I suggest we remove people from the `CODEOWNERS` file if we don't see them in ~3 months or so. That will help keep the list of reviewers focused and it will avoid the situation where everybody thinks someone else will deal with the review.

**Comment by henrif75** [comment]
I agree @mgeisler with cleaning up the `CODEOWNERS` file (for all languages). I'll get the stats to give a first pass.

File: po/zh-CN.po [hunk 1]
```diff
@@ -2024,7 +2024,7 @@ msgstr ""
 #: src/welcome-day-1.md:24
 msgid ""
 "The questions will likely mean that we talk about things ahead of the slides."
-msgstr "有些问题会导致我们提前谈到后面的内容"
+msgstr "有些问题会导致我们提前谈到后面的内容."
```
Comment (kongy) [inline po/zh-CN.po:2027]: use "。"

File: po/zh-CN.po [hunk 2]
```diff
@@ -2054,7 +2054,7 @@ msgstr ""
 
 #: src/welcome-day-1.md:36
 msgid "Morning: 9:00 to 12:00,"
-msgstr "早上：9:00 到 12:00，"
+msgstr "早上：9:00 到 12:00,"
```
Comment (kongy) [inline po/zh-CN.po:2057]: Why change to English comma?

Comment (SketchK) [reply po/zh-CN.po:2057]: image

If you use Poedit locally, you'll be prompted to make this change, or you'll be warned of an error!

File: po/zh-CN.po (hunk 1 above)
Comment (SketchK) [reply po/zh-CN.po:2027]: image

If you use Poedit locally, you'll be prompted to make this change, or you'll be warned of an error!

Comment (mgeisler) [reply po/zh-CN.po:2027]: Ah, interesting! That looks like a problem with Poedit and we should ignore the error in this case.

I created https://github.com/vslavik/poedit/issues/809 for this in the Poedit repository. Perhaps they have a workaround.

File: po/zh-CN.po [hunk 3]
```diff
@@ -1914,7 +1914,7 @@ msgid ""
 "example, using the example on the previous page, make `src/main.rs` look like"
 msgstr ""
 "将 `src/main.rs` 中的样板代码替换为你自己的代码。例如， 使用上一页中的示例，"
-"将 `src/main.rs` 改为："
+"将 `src/main.rs` 改为这样"
```
Comment (NoahDragon) [inline po/zh-CN.po:1917]: If this change is also caused by the poedit warning, I'd suggest to keep the previous transaltion which would be fit to the context.

File: po/zh-CN.po [hunk 4]
```diff
@@ -19387,7 +19387,7 @@ msgstr ""
 
 #: src/glossary.md:72
 msgid "mutable:"
-msgstr "可变"
+msgstr "可变:"
```
Comment (NoahDragon) [inline po/zh-CN.po:19390]: This change is inconsistent with others, like line 19382 and 19377. I agree this need to be changed, but it is better to change all related in a bulk.

File: po/zh-CN.po [hunk 5]
```diff
@@ -540,7 +540,7 @@ msgstr "结构化错误处理"
 
 #: src/SUMMARY.md:165
 msgid "Propagating Errors with ?"
-msgstr "使用 ? 传播错误"
+msgstr "传播错误用的 ？"
```
[the remark this answers is not in the document]
Comment (SketchK) [reply po/zh-CN.po:543]: This is Poedit‘s suggestion! @NoahDragon 

image

File: po/zh-CN.po (hunk 3 above)
Comment (SketchK) [reply po/zh-CN.po:1917]: This is Poedit's original suggestion
image

File: po/zh-CN.po (hunk 4 above)
Comment (SketchK) [reply po/zh-CN.po:19390]: @NoahDragon Yes, This commit is to resolve Poedit's warning, so you may wander why,  but all of it is suggestion of Poedit

image

File: po/zh-CN.po (hunk 1 above)
Comment (mgeisler) [reply po/zh-CN.po:2027]: Hi @SketchK, could you reply on https://github.com/vslavik/poedit/issues/809 to explain what you did to get the warning?

File: po/zh-CN.po (hunk 2 above)
Comment (mgeisler) [reply po/zh-CN.po:2057]: @SketchK, let's ignore the warning — if you find a way to turn it off, please help us document it.

File: po/zh-CN.po (hunk 1 above)
Comment (SketchK) [reply po/zh-CN.po:2027]: > Hi @SketchK, could you reply on [vslavik/poedit#809](https://github.com/vslavik/poedit/issues/809) to explain what you did to get the warning?

OK, Sir, I write something in that issue @mgeisler

File: po/zh-CN.po (hunk 4 above)
Comment (NoahDragon) [reply po/zh-CN.po:19390]: Thank you! I would recommend to separate the Poedit linter check warning fix into another PR. These warnings are not really useful for Chinese translation, and have many discussion already:
* https://stackoverflow.com/questions/44644152/how-to-hide-silly-warnings-in-poedit
* https://github.com/vslavik/poedit/issues/395
* https://github.com/vslavik/poedit/issues/445

@mgeisler Do you think if it is okay to ingore the linter warnings?

File: po/zh-CN.po (hunk 1 above)
Comment (mgeisler) [reply po/zh-CN.po:2027]: @SketchK, please apply the suggestion by @kongy above so we can resolve this comment.

File: po/zh-CN.po (hunk 5 above)
[the remark this answers is not in the document]
Comment (mgeisler) [reply po/zh-CN.po:543]: Feel free to ignore the warning in Poedit:

image

:smile: 

The only rules we have are encoded in the GitHub checks on the PR — the advice from Poedit should be taken as hints and a help.

File: po/zh-CN.po (hunk 4 above)
Comment (mgeisler) [reply po/zh-CN.po:19390]: Thanks @NoahDragon for looking this up! I would be very happy to have language-specific advice in STYLE or TRANSLATIONS or somewhere else — whatever feels most natural to the language communities.

**Comment by mgeisler** [comment]
Hey @SketchK, I removed myself and @henrif75 as reviewers — we're not actually qualified to skim through the Chinese here :smile: I think you've already gotten some great actionable feedback from @NoahDragon and @kongy and you should just apply those fixes.

When that is done, one of them will approve the PR and we can merge it. Thanks for all the work you do here!

File: po/zh-CN.po (hunk 2 above)
Comment (mgeisler) [reply po/zh-CN.po:2057]: Please edit it back to use `，` and just ignore the warning for now.

**Comment by mgeisler** [comment]
Hi @SketchK, will you resubmit this as another PR? I had the impression that the changes were good (based on the reviews of @kongy and @NoahDragon), so I would be sad to lose them!

**Comment by SketchK** [comment]
This PR just fix Poedit warning.
It looks like @NoahDragon  advised me to ignore the warnings, which shouldn't seem to be very useful, so I turned him off。

**Comment by mgeisler** [comment]
> @mgeisler This PR just fix Poedit warning. It looks like @NoahDragon advised me to ignore the warnings, which shouldn't seem to be very useful, so I turned him off。

Oh, I understand now! Thanks for the explanation.

**Pull request closed without merging**